### PR TITLE
Fix error message when parsing unknown JSON enum value

### PR DESCRIPTION
**Pull request by willhbr**
`#string_value` can be the value of the next token, which results in a confusing error message.

```crystal
require "json"

enum Test
  Valid
  Other
end

struct Foo
  include JSON::Serializable
  @foo : Test
end

text = %({"foo": "value", "things": "Stuff"})

Foo.from_json(text)
```

On `1.9.2` this will raise an unhelpful error:

```
$ crystal run test.cr
Unhandled exception: Unknown enum Test value: "things" at line 1, column 26
  parsing Foo#foo at line 1, column 2 (JSON::SerializableError)
  from /usr/share/crystal/src/json/serialization.cr:182:7 in 'initialize:__pull_for_json_serializable'
```

Note that if it's not part of a larger JSON object, the message will be correct:

```crystal
Test.from_json(%("value") # => Unhandled exception: Unknown enum Test value: "value"
```

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

**Comment by willhbr** [comment]
Is there anything else I should do here before it gets merged?

**Comment by straight-shoota** [comment]
No, we're just waiting for a second approval from a Core Team member (see https://github.com/crystal-lang/crystal/blob/master/CONTRIBUTING.md#accepting-a-pull-request).

And please do not force push to a PR branch. That unnecessarily complicates reviewing due to lost history (see https://github.com/crystal-lang/crystal/blob/master/CONTRIBUTING.md#making-good-pull-requests).

**Comment by willhbr** [comment]
Ah sorry, I'm used to doing endless rebases and hadn't seen that part of the contributing guide. Thanks for pointing that out.

**Review by beta-ziliani — APPROVED** [review]

**Pull request merged**
